### ISSUE TITLE
Option to deliver images via Drupal, not directly

### ISSUE THREAD

**Issue by jurgenhaas** [wodby/drupal-nginx#29]
Currently, assets are delivered directly by Nginx because of this location:

```
        location ~* ^.+\.(?:css|cur|js|jpe?g|gif|htc|ico|png|xml|otf|ttf|eot|woff|woff2|svg|svgz)xxx$ {
            access_log off;
            expires 30d;
            tcp_nodelay off;
            open_file_cache max=3000 inactive=120s;
            open_file_cache_valid 45s;
            open_file_cache_min_uses 2;
            open_file_cache_errors off;

            location ~* ^.+\.svgz$ {
                gzip off;
                add_header Content-Encoding gzip;
            }
        }
```

It would be great to get an environment variable that allows to disable that for images and deliver those by @drupal instead.

Use case: we are using docker4drupal in development environments and together with a module called stage_file_proxy it is possible to grab all the images from the live site instead of copying them all to each of the dev environments. But that only works if Drupal gets involved in delivering the images, otherwise it can't react to the request.

**Comment by rsmylski** [comment]
Our company does the same thing, and this would be a wonderful option to allow control over nginx vs drupal for all media, probably everything in that list except css/js), not just images.

**Comment by elalemanyo** [comment]
Hi,
I do something similar but we don't use any extra module, we just add this to nginx settings:

```
 ## Try local files first, otherwise redirect to a development server.
 location ^~ /sites/default/files {
   try_files $uri @dev_redirect;
 }
 ## Development Server Redirect
 location @dev_redirect {
   rewrite ^ http://dev.example.com$request_uri? permanent;
 }
```
Idea:
- https://www.lullabot.com/articles/using-remote-image-files-when-you-develop-locally
- http://dropbucket.org/node/337


Maybe we could add a new environment variable where the live server url could be set.

Thanks!

**Comment by csandanov** [comment]
Thanks, @elalemanyo, it's a great idea. See referenced commit – we now have `NGINX_DRUPAL_FILE_PROXY_URL` to specify URL to the Drupal instance with files, e.g. `http://dev.example.com` from your example

**Comment by elalemanyo** [comment]
@csandanov Did you try it? I did it and is not working, images are not being redirected so I get lots of 404. Did I miss something? I just add the NGINX_DRUPAL_FILE_PROXY_URL variable.

Thanks for the fast implementation!

**Comment by rsmylski** [comment]
@csandanov I'm seeing the same results as @elalemanyo - 404s for media.

I still think that a flag for whether we serve assets from nginx or drupal should be the way to go. This allows for using stage_file_proxy to get images from another server, but if you upload new local images, it will load those from your local as well.

Ultimately it's the user who controls how the media is loading, so if it is slower using drupal, that is the user's choice

**Comment by elalemanyo** [comment]
@rsmylski With the suggested nginx settings you can still upload new local images. I think is always better to solve the "problem" without needing an extra module.
@csandanov Maybe the problems is that the @file_proxy is at the end of drupal.conf and needs to be a little bit earlier. Maybe after the "location ~* /system/files/" block?

**Comment by csandanov** [comment]
Closed by mistake via commit message. Did it work for you guys?

**Comment by elalemanyo** [comment]
@csandanov Yes! it is working great!

Thanks

**Comment by jurgenhaas** [comment]
TBH I wasn't quite sure how to test it as it isn't in any of the released docker images yet, is it? Do I have to build that myself or what would be the best approach here?

**Comment by elalemanyo** [comment]
Just add NGINX_DRUPAL_FILE_PROXY_URL variable to your nginx service.
For example something like this is working for me:
```
nginx:
    image: wodby/drupal-nginx:7-1.13
    environment:
        NGINX_DRUPAL_FILE_PROXY_URL: http://live_server.com
```

**Comment by jurgenhaas** [comment]
Ah, OK. So this feature has already been there, it just wasn't documented. That's what I was missing.

I've now tested this and can confirm that it is working great !

**Comment by csandanov** [comment]
Released with [docke4drupal 2.5.0](https://github.com/wodby/docker4drupal/releases/tag/2.5.0) and [documented](https://docker4drupal.readthedocs.io/en/latest/containers/nginx/#files-proxy)
